tokio/src/time/interval.rs

```diff
@@ -103,26 +109,159 @@ pub fn interval_at(start: Instant, period: Duration) -> Interval {
+///     }
+/// }
+/// ```
+///
+/// Generally, a tick is missed if someone spends too much time without calling
+/// [`tick()`](Interval::tick).
+///
+/// By default, when at tick is missed, [`Interval`] fires ticks as quickly as
+/// it can until it is back to where it should be. However, this is not always
+/// the desired behavior. `MissedTickBehavior` allows users to specify which
+/// behavior they want [`Interval`] to exhibit. Each variant represents a
+/// different strategy.
+///
+/// Note that because the executor cannot guarantee exact precision with timers,
+/// these strategies will only apply when the error in time is greater than 5
+/// milliseconds.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum MissedTickBehavior {
+    /// Tick as fast as possible until caught up.
+    ///
+    /// When this strategy is used, [`Interval`] schedules ticks "normally" (the
+    /// same as it would have if the ticks hadn't been delayed), which results
+    /// in ticks being fired as fast as it can until it is caught up in time to
+    /// where it should be.
+    ///
+    /// This looks something like this:
+    /// ```text
+    /// Expected ticks: |     1     |     2     |     3     |     4     |     5     |     6     |
+    /// Actual ticks:   | work -----|          delay          | work | work | work -| work -----|
+    /// ```
+    ///
+    /// This is the default behavior when [`Interval`] is created with
+    /// [`interval`] and [`interval_at`].
+    Burst,
+
+    /// Delay missed ticks to happen at multiples of `period` from when [`tick`]
+    /// was called.
+    ///
+    /// When this strategy is used, [`Interval`] schedules all future ticks to
+    /// happen at a regular `period` from the point when [`tick`] was called.
+    ///
+    /// This looks something like this:
+    /// ```text
+    /// Expected ticks: |     1     |     2     |     3     |     4     |     5     |     6     |
+    /// Actual ticks:   | work -----|          delay          | work -----| work -----| work -----|
+    /// ```
+    ///
+    /// Note that as a result, the ticks are no longer guaranteed to happen at
+    /// a multiple of `period` from `delay`.
+    ///
+    /// [`tick`]: Interval::tick
+    Delay,
+
+    /// Skip the missed ticks and tick on the next multiple of `period`.
+    ///
+    /// When this strategy is used, [`Interval`] schedules the next tick for the
+    /// closest multiple of `period` from when the [`Interval`] first ticked.
+    ///
+    /// This looks something like this:
+    /// ```text
+    /// Expected ticks: |     1     |     2     |     3     |     4     |     5     |     6     |
+    /// Actual ticks:   | work -----|          delay          | work ---| work -----| work -----|
+    // * Where `s` is the start time and `p` is the period
+    /// ```
+    ///
+    /// Note that the ticks aren't guarenteed to be one `period` away from the
+    /// last tick, but they will be a multiple of `period` away.
+    Skip,
```

Comment:
I don't think this is true. When we get a delay, the period is _not_ a multiple of the period. In fact, it's the `Delay` version that does this.

Reply:
I'm trying to figure out how to reword it, but I can't think of a good way to. I don't feel like those lines are necessary. What do you think?

Reply:
Removing it is fine.